src/TEdit/Terraria/World.FileV2.cs

```diff
@@ -165,6 +165,18 @@ public static int SaveTiles(Tile[,] tiles, int version, int maxX, int maxY, Bina
                 {
                     Tile tile = tiles[x, y];
 
+                    // check if target version allows tile/wall.
+                    if (tile.Type >= SaveConfiguration.SaveVersions[version].MaxTileId)
+                    {
+                        tile.U = 0;
+                        tile.IsActive = false;
+                    }
+                    if (tile.Wall >= SaveConfiguration.SaveVersions[version].MaxWallId)
+                    {
+                        tile.Wall = 0;
+                        tile.IsActive = false;

```

Comment:
This doesn't need to be here. Since there could be wall, headers, wires, etc. still present and this should only remove the wall if the ID is out of range.